Automattic/wp-calypso
# Jetpack Backup: add some additional Tracks events on Storage Retention settings and update `Learn more` link

Related to https://github.com/Automattic/jetpack-backup-team/issues/489

## Proposed Changes
* Add `calypso_jetpack_backup_storage_retention_view` to track when a user see the storage retention settings.
* Add `calypso_jetpack_backup_storage_retention_tooltip_open_click` to track when a user clicks to open the tooltip on the retention setting.
* Add `calypso_jetpack_backup_storage_retention_tooltip_learn_more_click` to track when a user clicks on the `Learn more` link on the retention setting tooltip.
* Add `calypso_jetpack_backup_storage_retention_confirmation_cancel_click` to track when a user clicks on `Cancel` button on the update retention confirmation modal.
* Update `Learn more` link to use `ExternalLink` from `@wordpress/components`. This is mostly to ensure it is treated as an external link and to include the respecting icon at the end:

| Before | After |
|---|---|
| ![CleanShot 2024-05-31 at 23 20 32@2x](https://github.com/Automattic/wp-calypso/assets/1488641/4b4b0373-9630-4aff-8d05-fc1c45551605) | ![CleanShot 2024-05-31 at 23 19 59@2x](https://github.com/Automattic/wp-calypso/assets/1488641/d20eedcd-8da2-43f8-b737-8977634706f6) |

## Why are these changes being made?


These specific Tracks events will help us validate the feature usage of the Storage Retention settings.

## Testing Instructions



* Spin up a Jetpack Cloud live branch
* Open the Network tab in your developer tools
* Navigate to Jetpack Cloud > Settings
  * Once the retention settings loads, you should see a request to `t.gif` with the `calypso_jetpack_backup_storage_retention_view` event
* Try clicking on the `(i)` icon to open the tooltip
  * You should see a request to `t.gif` with the `calypso_jetpack_backup_storage_retention_tooltip_open_click` event
* Ensure the `Learn more ↗` looks as shared above and then click it
  * You should see a request to `t.gif` with the `calypso_jetpack_backup_storage_retention_tooltip_learn_more_click` event
* Now try to update your retention setting to a lower value than the one you have. For example, if you already have a 30-days retention, try updating it to 7-days.
* When the confirmation modal opens, try clicking on `Cancel`
  * You should see a request to `t.gif` with the `calypso_jetpack_backup_storage_retention_confirmation_cancel_click` event

## Pre-merge Checklist
